dotCMS/src/main/java/com/dotcms/storage/RequestMetaData.java

```diff
@@ -0,0 +1,113 @@
+package com.dotcms.storage;
+
+import java.io.File;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+/**
+ * Encapsulates the parameters to request the metadata
+ * it could be by cache or file system
+ * @author jsanca
+ */
+public class RequestMetaData {
+
+    /**
+     * Provides the key for the storage
+     */
+    private final StorageKey    storageKey;
+
+    /**
+     * If true, means the medatada output will be stores in the memory cache.
+     */
+    private final boolean           cache;
+
+    /**
+     * Cache key supplier, if cache is true
+     */
+    private final Supplier<String>  cacheKeySupplier;
+
+    /**
+     * In case the metadata is retrieved from the storage instead of the cache,
+     * you can wrap the metadata recovery from the storage in order to add, mod or remove values
+     */
+    private final Function<Map<String, Object>, Map<String, Object>> wrapMetadataMapForCache;
+
+    private RequestMetaData(final Builder builder) {
+
+        this.cache                   = builder.cache;
+        this.cacheKeySupplier        = builder.cacheKeySupplier;
+        this.storageKey              = builder.storageKey;
+        this.wrapMetadataMapForCache = builder.wrapMetadataMapForCache;
+    }
+
+    public StorageKey getStorageKey() {
+        return storageKey;
+    }
+
+    public boolean isCache() {
+        return cache;
+    }
+
+    public Supplier<String> getCacheKeySupplier() {
+        return cacheKeySupplier;
+    }
+
+    public Function<Map<String, Object>, Map<String, Object>> getWrapMetadataMapForCache() {
+        return wrapMetadataMapForCache;
+    }
+
+    public static final class Builder {
+
+        /**
+         * Provides the key for the storage
+         */
+        private  StorageKey    storageKey;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Field storageKey has the same name as a method](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5547873)